drivers/gpio/gpio_ite_it8xxx2.c

```diff
@@ -522,6 +522,9 @@ static int gpio_ite_pin_interrupt_configure(const struct device *dev,
 		return -ENOTSUP;
 	}
 
+	/* Disable irq before configuring it */
+	irq_disable(gpio_irq);
```

Comment:
We could refactor this a bit
```
	/* Disable GPIO interrupt */
	irq_disable(gpio_irq);

	if (mode == GPIO_INT_MODE_DISABLED) {
		return 0;
	}
```
and remove the no longer needed code from the top of the function.